src/Transports/MassTransit.RabbitMqTransport/RabbitMqTransport/RabbitMqBasicConsumer.cs

```diff
@@ -270,29 +236,19 @@ protected override async Task StopAgent(StopContext context)
             }
         }
 
-        async Task CancelAndWaitForDeliveryComplete(StopContext context)
+        protected override async Task ActiveAndActualAgentsCompleted(StopContext context)
         {
             try
             {
-                await _model.BasicCancel(_consumerTag).ConfigureAwait(false);
+                if (IsGracefulShutdown)
```

Comment:
yeah, this seems like a StopProcessing, which if it is graceful, we'd tell the model to cancel this consumer.